animate: Add spin animation

soreau opened the pull request.

ammen99 commented on plugins/animate/spin.hpp:
```diff
@@ -0,0 +1,100 @@
+            ->get_transformer<wf::scene::view_2d_transformer_t>(spin_transformer_name);
+        auto progress = this->progression.progress();
+        transform->alpha   = progress;
+        transform->angle   = progress * M_PI * 4.0;
```
Idea: let the user configure the `4.0` constant, that way they can configure how often the window will rotate before closing/opening completely.

soreau replied:
Already way ahead of you, the [wayfire-animations branch](https://github.com/soreau/wayfire/blob/wayfire-animations/plugins/animate/spin.hpp#L82-L83) has a version which provides the feature you request.

ammen99 replied:
great :)

ammen99 reviewed (approved):
Just as the other animation, looks simple but works well! We can merge after you rebase. I have added one suggestion, feel free to ignore it if you don't like it.

soreau commented:
> Just as the other animation, looks simple but works well! We can merge after you rebase. I have added one suggestion, feel free to ignore it if you don't like it.

It's a good suggestion, I've implemented it. 👍

ammen99 reviewed (approved):
thanks :)
